### PR TITLE
Shared tabbed bar

### LINKED ISSUE #544: [UI Components] Tabbed Bars
Create a reusable tabbed bar component that will render a component based on the selected tab and that also allows for a small circle indicator

### PR DESCRIPTION
**Pull request by ivan-kishko**
## Description of the change

> Shared Tabbed Bar

## Type of change
Type: Non-breaking refactor 

> All pull requests must have at least one of the following labels applied (otherwise the PR will fail):

| Label                       	| Description                                                                                               	|
|-----------------------------	|-----------------------------------------------------------------------------------------------------------	|
| Type: Bug                   	| non-breaking change that fixes an issue                                                                   	|
| Type: Feature               	| non-breaking change that adds functionality                                                               	|
| Type: Breaking Change       	| fix or feature that would cause existing functionality to not work as expected                            	|
| Type: Non-breaking refactor 	| change addresses some tech debt item or prepares for a later change, but does not change functionality    	|
| Type: Configuration Change  	| adjusts configuration to achieve some end related to functionality, development, performance, or security 	|
| Type: Dependency Upgrade      | upgrades a project dependency - these changes are not included in release notes                             |

## Related issues

Closes #544 

### Development

**This box MUST be checked by the submitter prior to merging**:
- [x] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: common/components/Dropdown/Dropdown.tsx [hunk 1]
```diff
@@ -62,7 +62,7 @@ export function Dropdown({
   alignment,
   overflow,
   fullWidth,
-}: Props) {
+}: DropdownProps) {
```
Comment (mxosman) [inline common/components/Dropdown/Dropdown.tsx:65]: Thank you for the specificity!

File: common/components/TabbedBar/TabbedBar.styled.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    return `padding: 8px 0; ${typography.sizeCSS.normal};`;
+  }}
+  
+  &:hover {
+    cursor: pointer;
+    color: ${({ enabled, selected }) => {
+      if (enabled !== undefined)
+        return (
+          !selected && (enabled ? palette.solid.blue : palette.solid.darkgrey)
+        );
```
Comment (mxosman) [inline common/components/TabbedBar/TabbedBar.styled.tsx:57]: Could we do:

```
      if (enabled !== undefined && !selected)
        return enabled ? palette.solid.blue : palette.solid.darkgrey
        );
```

Maybe even simplify the whole thing L54-58:
```
      return (!enabled && !selected) palette.solid.darkgrey : palette.solid.blue
```

I could be wrong - haven't double checked this!

**Review by mxosman — APPROVED** [review]
This is awesome, Ivan! Thank you for flying through these. Left one super tiny nit suggestion if it works - otherwise, this is great!

File: common/components/TabbedBar/TabbedBar.styled.tsx (hunk 2 above)
Comment (ivan-kishko) [reply common/components/TabbedBar/TabbedBar.styled.tsx]: [e1ffcdc](https://github.com/Recidiviz/justice-counts/pull/605/commits/e1ffcdc3ed6b6aa23166af7f59f676b9affae45d)

Comment (mxosman) [reply common/components/TabbedBar/TabbedBar.styled.tsx]: Great call - thanks for double checking!

**Pull request merged**